plugin/keyring_vault/tests/mtr/mount_point_service.inc

```diff
@@ -0,0 +1,62 @@
+# Creates or deletes secret mount point specified in keyring_vault configuration file
+# Mount point is :
+#  - created when MOUNT_POINT_SERVICE_OP is set to CREATE
+#  - deleted when MOUNT_POINT_SERVICE_OP is set to DELETE
+
+# The sourcing test needs to set $KEYRING_CONF_FILE variable to
+# the location of keyring_vault configuration file and
+# MOUNT_POINT_SERVICE_OP variable to CREATE or DELETE
+let KEYRING_CONF_FILE=$KEYRING_CONF_FILE;
+let SERVER_UUID= query_get_value(SELECT @@SERVER_UUID, @@SERVER_UUID, 1);
+let MOUNT_POINT_SERVICE_OP=$MOUNT_POINT_SERVICE_OP;
+
+--perl
+  use strict;
+  use MIME::Base64 qw( decode_base64 );
+  my $keyring_conf_file= $ENV{'KEYRING_CONF_FILE'} or die("KEYRING_CONF_FILE not set\n");
+  my $server_uuid= $ENV{'SERVER_UUID'} or die("SERVER_UUID not set\n");
+  my $mount_point_service_op=$ENV{'MOUNT_POINT_SERVICE_OP'};
+  my $token;
+  my $vault_url;
+  my $secret_mount_point;
+  my $vault_ca;
+  my $CONF_FILE;
+  open(CONF_FILE, "$keyring_conf_file") or die("Could not open configuration file.\n");
+  while (my $row = <CONF_FILE>)
+  {
+    if ($row =~ m/token[ ]*=[ ]*(.*)/)
+    {
+      $token=$1;
+    }
+    elsif ($row =~ m/vault_url[ ]*=[ ]*(.*)/)
+    {
+      $vault_url=$1;
+    }
+    elsif ($row =~ m/secret_mount_point[ ]*= [ ]*(.*)/)
+    {
+      $secret_mount_point=$1;
+    }
+    elsif ($row =~ m/vault_ca[ ]*= [ ]*(.*)/)
+    {
+      $vault_ca=$1;
+    }
+  }
+  close(CONF_FILE);
+  if ($token eq "" || $vault_url eq "" || $secret_mount_point eq "")
```

Comment:
`|| $vault_ca eq ""`? Or is it optional?

Reply:
it's optional

Reply:
Then you need to take into account this when you form `curl` command lines for both `CREATE` and `DELETE`
```perl
if($vault_ca)
{
    system(qq#curl -H "X-Vault-Token: $token" --cacert $vault_ca -X DELETE $vault_url/v1/sys/mounts/$secret_mount_point#);
}
else
{
    system(qq#curl -H "X-Vault-Token: $token" -X DELETE $vault_url/v1/sys/mounts/$secret_mount_point#);
}
```

Reply:
Done